android/build.gradle

```diff
@@ -1,3 +1,38 @@
+import groovy.json.JsonSlurper
+
+def safeExtGet(prop, fallback) {
+    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+}
+
+def warnNoAutoInit() {
+    logger.warn("Couldn't auto-initialize the Heap library; manually initialize it by following the instructions at https://docs.heap.io/docs/react-native.")
+}
+
+def warnDisabled() {
+    logger.warn("Not auto-initializing the Heap library because it was manually disabled; set `enabled` to `true` in `heap.json` to enable auto-initialization.")
+}
+
+ext.HEAP_CONFIG_FILENAME = 'heap.config.json'
+ext.HEAP_APP_ID_FIELD_NAME_DEV = 'HEAP_APP_ID_DEV'
+ext.HEAP_APP_ID_FIELD_NAME_PROD = 'HEAP_APP_ID_PROD'
+
+def heapAppId(buildType) {
+    def file = rootProject.file("../${HEAP_CONFIG_FILENAME}")
+    if (!file.exists()) { return warnNoAutoInit() }
+
+    def jsonSlurper = new JsonSlurper()
+    def data = jsonSlurper.parseText(file.text.trim())[buildType]
+    def defaultData = jsonSlurper.parseText(file.text.trim())["default"]
+
+    // KLUDGE: Can't use the elvis operator because we want to treat `null` and `false` differently.
+    def autoInit = data?.heapAutoInit
+    if (autoInit == null) { autoInit = defaultData?.heapAutoInit }
+    if (autoInit == null) { autoInit = true }
+
+    if (!autoInit) { return warnDisabled() }
+    return data?.heapAppId ?: defaultData?.heapAppId ?: warnNoAutoInit();
```

Comment:
Now that I'm looking at it again, this seems a little smelly... like we're either returning an app id (string) or we're... printing a warning??  Halting in the middle of the process??  Is there an exception that should be thrown or some other way to indicate what's gone wrong?

Reply:
The contract here is that a `null` return value indicates that an app id could not be found. I'm not a big fan of using exceptions for predictable errors.